crates/dojo/bindgen/src/plugins/unrealengine/mod.rs

```diff
@@ -0,0 +1,1376 @@
+
+    void ParseModelsAndSend(struct CArrayStruct *models);
+
+    void ExecuteFromOutside(const FControllerAccount& account,
+                            const FString& to,
+                            const FString& selector,
+                            const FString& calldataParameter);
+
+    void ExecuteRawDeprecated(const FAccount& account,
+                              const FString& to,
+                              const FString& selector,
+                              const FString& calldataParameter);
+
+public:
+    AGeneratedHelpers();
+    ~AGeneratedHelpers();
+
+    AGeneratedHelpers* GetGlobalInstance();
+    void SetGlobalInstance(AGeneratedHelpers* instance);
+
+    UFUNCTION(BlueprintCallable)
+    void Connect(const FString& torii_url, const FString& world);
+
+    UFUNCTION(BlueprintCallable)
+    void FetchExistingModels();
+
+    UFUNCTION(BlueprintCallable)
+    void SubscribeOnDojoModelUpdate();
+
+    UFUNCTION(BlueprintCallable)
+    FAccount CreateAccountDeprecated(const FString& rpc_url,
+                                     const FString& address,
+                                     const FString& private_key);
+
+    UFUNCTION(BlueprintCallable)
+    FAccount CreateBurnerDeprecated(const FString& rpc_url,
+                                    const FString& address,
+                                    const FString& private_key);
+
+    DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnDojoModelUpdated, UDojoModel*, Model);
+
+    UPROPERTY(BlueprintAssignable)
+    FOnDojoModelUpdated OnDojoModelUpdated;
+
+    // CONTROLLER
+    UFUNCTION(BlueprintCallable)
+    void ControllerGetAccountOrConnect(const FString& rpc_url, const FString& chain_id);
+
+    DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnDojoControllerAccount,
+                                                struct FControllerAccount,
+                                                Account);
+
+    UPROPERTY(BlueprintAssignable)
+    FOnDojoControllerAccount FOnDojoControllerAccount;
+
+    // CALLS
+    {}
+}};",
+            parse_models_functions, functions_calls
+        ));
+
+        out
+    }
+
+    fn generate_parse_models_functions(models: &[(&String, &DojoModel)]) -> String {
+        let parse_models_functions_bodies = models
+            .iter()
+            .map(|(_, model)| {
+                format!(
+                    "UDojoModel* AGeneratedHelpers::parse{namespace}{model_name}Model(struct \
+                     Struct* model)
+{{
+    UDojoModel{namespace}{model_name}* Model = NewObject<UDojoModel{namespace}{model_name}>();
+    CArrayMember* members = &model->children;
+
+    for (int k = 0; k < members->data_len; k++) {{
+        Member* member = &members->data[k];
+        {process_members}
+    }}
+
+    FDojoModule::CArrayFree(members->data, members->data_len);
+    return Model;
+}}
+",
+                    namespace = Self::to_pascal_case(&get_namespace_from_tag(&model.tag)),
+                    model_name = model
```

Comment:
Even if it's not a critical code, we may extract some temporary variable inside the map to avoid iterating several times on the same data structure.